enterprise/jmx-monitoring/src/test/java/io/crate/beans/QueryStatsTest.java

```diff
@@ -61,26 +64,67 @@ private Metrics createMetric(Classification classification, long duration) {
+        assertThat(metricsByCommand.get(StatementType.DELETE), is(notNullValue()));
+        assertThat(metricsByCommand.get(StatementType.DDL), is(notNullValue()));
+        assertThat(metricsByCommand.get(StatementType.MANAGEMENT), is(notNullValue()));
+        assertThat(metricsByCommand.get(StatementType.COPY), is(notNullValue()));
+        assertThat(metricsByCommand.get(StatementType.ALL), is(notNullValue()));
+
+        assertThat(metricsByCommand.get(StatementType.UNDEFINED), is(nullValue()));
+    }
+
     @Test
     public void testCreateMetricsMap() {
         Map<StatementType, QueryStats.Metric> metricsByCommand = createMetricsMap(metrics, Collections.emptyMap(), 2000, 0);
         assertThat(metricsByCommand.size(), is(6));
 
+        assertThat(metricsByCommand.get(StatementType.SELECT).totalCount(), is(2L));
+        assertThat(metricsByCommand.get(StatementType.SELECT).sumOfDurations(), is(70L));
         assertThat(metricsByCommand.get(StatementType.SELECT).avgDurationInMs(), is(35.0));
         assertThat(metricsByCommand.get(StatementType.SELECT).statementsPerSec(), is(1.0));
 
+        assertThat(metricsByCommand.get(StatementType.INSERT).totalCount(), is(1L));
+        assertThat(metricsByCommand.get(StatementType.INSERT).sumOfDurations(), is(19L));
         assertThat(metricsByCommand.get(StatementType.INSERT).avgDurationInMs(), is(19.0));
         assertThat(metricsByCommand.get(StatementType.INSERT).statementsPerSec(), is(0.5));
 
+        assertThat(metricsByCommand.get(StatementType.UPDATE).totalCount(), is(1L));
+        assertThat(metricsByCommand.get(StatementType.UPDATE).sumOfDurations(), is(20L));
         assertThat(metricsByCommand.get(StatementType.UPDATE).avgDurationInMs(), is(20.0));
         assertThat(metricsByCommand.get(StatementType.UPDATE).statementsPerSec(), is(0.5));
 
+        assertThat(metricsByCommand.get(StatementType.DELETE).totalCount(), is(2L));
+        assertThat(metricsByCommand.get(StatementType.DELETE).sumOfDurations(), is(15L));
         assertThat(metricsByCommand.get(StatementType.DELETE).avgDurationInMs(), is(7.5));
         assertThat(metricsByCommand.get(StatementType.DELETE).statementsPerSec(), is(1.0));
 
-        assertThat(metricsByCommand.get(StatementType.UNDEFINED).avgDurationInMs(), is(1.0));
-        assertThat(metricsByCommand.get(StatementType.UNDEFINED).statementsPerSec(), is(0.5));
+        assertThat(metricsByCommand.get(StatementType.DDL).totalCount(), is(1L));
+        assertThat(metricsByCommand.get(StatementType.DDL).sumOfDurations(), is(1L));
+        assertThat(metricsByCommand.get(StatementType.DDL).avgDurationInMs(), is(1.0));
+        assertThat(metricsByCommand.get(StatementType.DDL).statementsPerSec(), is(0.5));
 
+        assertThat(metricsByCommand.get(StatementType.ALL).totalCount(), is(7L));
+        assertThat(metricsByCommand.get(StatementType.ALL).sumOfDurations(), is(125L));
```

Comment:
afaik, the ALL type totalCount + sumOfDurations in never exposed, so we won't have to test it

Reply:
but we're not testing what the bean is exposing here, but what internal map is containing, so I don't particularly think it hurts to test it (especially as it affects the size of the map)